core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -138,11 +142,18 @@ public List<TaskInfo> getAllTaskInfo(@Context UriInfo uriInfo)
     @Produces(MediaType.APPLICATION_JSON)
     public void createOrUpdateTask(
             @PathParam("taskId") TaskId taskId,
+            @QueryParam("instanceId") String expectedInstanceId,
             TaskUpdateRequest taskUpdateRequest,
             @Context UriInfo uriInfo,
             @Suspended AsyncResponse asyncResponse)
     {
         requireNonNull(taskUpdateRequest, "taskUpdateRequest is null");
+        if (!nodeInstanceId.equals(expectedInstanceId)) {
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST)
```

Comment:
or should this be a TrinoException? cc @losipiuk

Reply:
You can resume with an exception and rely on the default mapper

Reply:
the question is whether we want fail fast or let the caller retry and do something else
i think it ultimately may not matter, so i think i will leave this as is for now.